app/tests/hanging_protocols_tests/test_forms.py

```diff
@@ -294,3 +312,415 @@ def test_hanging_protocol_clientside():
+                'Refer to the <a href="https://testserver/documentation/viewer-content/">documentation</a> for more information'
+            ),
+        ),
+        (
+            0,
+            1,
+            0,
+            1,
+            (
+                "The following interfaces are used in your {}: test-ci-overlay-0 and test-ci-undisplayable-0. "
+                "No interfaces of type image, chart, pdf, mp4, thumbnail_jpg or thumbnail_png are used. At least one interface of those types is needed to configure the viewer. "
+                'Refer to the <a href="https://testserver/documentation/viewer-content/">documentation</a> for more information'
+            ),
+        ),
+        (
+            1,
+            1,
+            1,
+            1,
+            (
+                "The following interfaces are used in your {}: test-ci-isolated-0, test-ci-image-0, test-ci-overlay-0, and test-ci-undisplayable-0. "
+                'Example usage: {{"main": ["test-ci-isolated-0"], "secondary": ["test-ci-image-0", "test-ci-overlay-0"]}}. '
+                'Refer to the <a href="https://testserver/documentation/viewer-content/">documentation</a> for more information'
+            ),
+        ),
+    ),
+)
+@pytest.mark.parametrize(
+    "object_factory,form_class",
+    (
+        (ArchiveItemFactory, ArchiveForm),
+        (DisplaySetFactory, ReaderStudyUpdateForm),
+    ),
+)
+@pytest.mark.django_db
+def test_archive_and_reader_study_forms_view_content_help_text(
+    number_of_images,
+    number_of_overlays,
+    number_of_isolated_interfaces,
+    number_of_undisplayable_interfaces,
+    expected_help_text,
+    object_factory,
+    form_class,
+):
+    civ_list = []
+
+    for i in range(number_of_isolated_interfaces):
+        civ = ComponentInterfaceValueFactory(
+            interface=ComponentInterfaceFactory(
+                kind=InterfaceKindChoices.CHART,
+                title=f"test-ci-isolated-{i}",
+            ),
+        )
+        civ_list.append(civ)
+
+    for i in range(number_of_images):
+        civ = ComponentInterfaceValueFactory(
+            interface=ComponentInterfaceFactory(
+                kind=InterfaceKindChoices.IMAGE,
+                title=f"test-ci-image-{i}",
+            ),
+        )
+        civ_list.append(civ)
+
+    for i in range(number_of_overlays):
+        civ = ComponentInterfaceValueFactory(
+            interface=ComponentInterfaceFactory(
+                kind=InterfaceKindChoices.SEGMENTATION,
+                title=f"test-ci-overlay-{i}",
+            ),
+        )
+        civ_list.append(civ)
+
+    for i in range(number_of_undisplayable_interfaces):
+        civ = ComponentInterfaceValueFactory(
+            interface=ComponentInterfaceFactory(
+                kind=InterfaceKind.InterfaceKindChoices.ZIP,
+                title=f"test-ci-undisplayable-{i}",
+            ),
+        )
```

Comment:
This could be a helper function that you re-use in all 3 tests.

Reply:
ah, great idea!
Actually, I made two fixtures, one to create a list of component interfaces `ci_list` (needed for Algorithm and Phase) and the second to create a list of component interface values `civ_list` (needed for Archive and ReaderStudy). That made the tests much cleaner and concise :)
The last function `test_generate_view_content_example` does not create a list but create the actual objects and the fixture is not needed there.

Done.